core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -112,6 +112,36 @@
         return null;
     }
 
+    /**
+     * Returns the value mapped to the given {@link AttributeKey} or {@code null} if there's no value set by
+     * {@link #setAttr(AttributeKey, Object)} or {@link #setAttrIfAbsent(AttributeKey, Object)}.
+     */
+    @Nullable
+    <V> V attr(AttributeKey<V> key);
+
+    /**
+     * Associates the specified value with the given {@link AttributeKey} in this context.
+     * If this context previously contained a mapping for the {@link AttributeKey},
+     * the old value is replaced by the specified value. Set {@code null} not to iterate the mapping from
+     * {@link #attrs()}.
+     */
+    <V> void setAttr(AttributeKey<V> key, @Nullable V value);
+
+    /**
+     * Associates the specified value with the given {@link AttributeKey} in this context only
+     * if this context does not contain a mapping for the {@link AttributeKey}.
+     *
+     * @return {@code null} if there was no mapping for the {@link AttributeKey} or the old value if there's
+     *         a mapping for the {@link AttributeKey}.
+     */
+    @Nullable
+    <V> V setAttrIfAbsent(AttributeKey<V> key, V value);
```

Comment:
Could we also have `computeAttrIfAbsent()`?

Reply:
Added!